native/explorer/src/dataframe.rs

```diff
@@ -282,20 +282,67 @@ pub fn df_sample_frac(
     Ok(ExDataFrame::new(new_df))
 }
 
+/// NOTE: The '_ref' parameter is needed to prevent the BEAM GC from collecting the stream too soon.
```

Comment:
Just for future reference, the way I plan to "fix" this is by making the `get_stream_pointer` receive a callback and you can access and use the pointer only within the callback. :)